datastorages/cmis/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -657,14 +658,22 @@ protected MetadataResourceExternalManagementProperties getMetadataResourceExtern
     ) {
         String metadataResourceExternalManagementPropertiesUrl = cmisConfiguration.getExternalResourceManagementUrl();
         if (!StringUtils.isEmpty(metadataResourceExternalManagementPropertiesUrl)) {
+            // {objectid}  objectId // It will be the type:visibility:metadataId:version:resourceId in base64
+            // i.e. folder::100::100                     # Folder in resource 100
+            // i.e. document:public:100:v1:sample.jpg    # public document 100 version v1 name sample.jpg
+            if (metadataResourceExternalManagementPropertiesUrl.contains("{objectid}")) {
+                metadataResourceExternalManagementPropertiesUrl = metadataResourceExternalManagementPropertiesUrl.replaceAll("(\\{objectid\\})",
+                    getResourceManagementExternalPropertiesObjectId((type == null ? "document" : (type instanceof Folder ? "folder" : "document")), visibility, metadataId, version, resourceId));
+            }
+
             // {id}  id
             if (metadataResourceExternalManagementPropertiesUrl.contains("{id}")) {
                 metadataResourceExternalManagementPropertiesUrl = metadataResourceExternalManagementPropertiesUrl.replaceAll("(\\{id\\})", (resourceId==null?"":resourceId));
             }
-            // {type:folder:document} // If the type is folder then type "folder" will be displayed else if document then "document" will be displayed
+            // {type} // If the type is folder then type "folder" will be displayed else if document then "document" will be displayed
             if (metadataResourceExternalManagementPropertiesUrl.contains("{type:")) {
-                metadataResourceExternalManagementPropertiesUrl = metadataResourceExternalManagementPropertiesUrl.replaceAll("\\{type:([a-zA-Z0-9]*?):([a-zA-Z0-9]*?)\\}",
-                    (type==null?"":(type instanceof Folder?"$1":"$2")));
+                metadataResourceExternalManagementPropertiesUrl = metadataResourceExternalManagementPropertiesUrl.replaceAll("(\\{type\\})",
+                    (type == null ? "document" : (type instanceof Folder ? "folder" : "document")));
```

Comment:
Previously it was used `"$1":"$2"` instead of `"folder" : "document"`. To check that this can not break existing setups?

Reply:
Yes this would have broken existing setup. 

The problem with this type is that it could be set to return custom values for the type but this was inconsistent with the type being returned in the {objectId} which was a hard coded value.

I had removed the old logic to no longer allow for custom name as I thought it could be confusing.

Looking at the code, I noticed that the old code was looking for "{type:" while the new code is supposed to looks for {type}

As they are different - I readded the old one so that it remains unchanged and I added the new {type} to be consistent with other providers.

Reply:
I reverted the changes and added a new one so it will not break any existing setup.